opentelemetry/src/baggage.rs

```diff
@@ -282,10 +282,29 @@ impl FromIterator<KeyValueMetadata> for Baggage {
     }
 }
 
+fn encode(s: Cow<'_, str>) -> String {
+    let special_characters = ['.', '-', '_', '~'];
+    let mut encoded_string = String::new();
+
+    for char in s.as_ref().chars() {
+        if char.is_ascii_alphanumeric() || special_characters.contains(&char) {
+            encoded_string.push(char);
+        } else if char == ' ' {
+            encoded_string.push_str("%20");
+        } else {
+            // Convert to a two-digit hexadecimal representation
+            for byte in char.to_string().as_bytes() {
```

Comment:
That seems inefficient, converting char to string, and then to bytes. Instead, can we directly iterate over the original string as bytes?

Reply:
yeah converting to char makes it easy to encode non ascii chars which could be multi byte.

Reply:
I meant something like this (not tested):
```rust
  for byte in s.as_bytes() {
        match *byte {
                    //alphanumeric...
                   b'a'.. b'z' | b 'A'.. b'Z' | ... => encoded_string.push(*byte as char);
                   b' ' =>  encoded_string.push_str("%20"),
                    //percentile encode all other type
                   _  =>  encoded_string.push_str(&format!("%{:02X}", byte)),
        }
  }
```

Reply:
hmm, let me try this out.